File: numbers/easy/java_solutions/PowerOfThree_326.java
package numbers.easy.java_solutions;

public class PowerOfThree_326 {

    public boolean isPowerOfThree(int n) {
        if (n < 1) {
            return false;
        }
        while (n > 1) {
            if (n % 3 != 0) {
                return false;
            }
            n /= 3;
        }
        return true;
    }

    public boolean isPowerOfThreeMax(int n) {
        return n > 0 && Math.pow(3, 19) % n == 0;
    }
}

File: leetcode/numbers/easy/golang_solutions/power_of_three_326.go
package golang_solutions

import "math"

func isPowerOfThree(n int) bool {
	if n < 1 {
		return false
	}
	for n > 1 {
		if n%3 != 0 {
			return false
		}
		n /= 3
	}
	return true
}

func isPowerOfThreeMax(n int) bool {
	return n > 0 && int(math.Pow(float64(3), float64(19)))%n == 0
}
